ui/src/app/pages/admin/admin-institution-edit.spec.tsx

```diff
@@ -514,64 +467,151 @@ describe('AdminInstitutionEditSpec - add mode', () => {
+    expect(findCTERARequired(wrapper).props.checked).toBeTruthy();
+
+    // change Registered to ADDRESS
+
+    expect(findRTAddress(wrapper).exists()).toBeFalsy();
+
+    await simulateComponentChange(wrapper, findRTDropdown(wrapper), InstitutionMembershipRequirement.ADDRESSES);
+
+    expect(findRTAddress(wrapper).exists()).toBeTruthy();
+    expect(findRTDomain(wrapper).exists()).toBeFalsy();
+
+    expect(findRTERARequired(wrapper).props.checked).toBeTruthy();
+    expect(findCTERARequired(wrapper).props.checked).toBeTruthy();
+  });
+
+  it('Should display error in case of invalid email Address Format in Registered Tier requirement', async() => {
+    const wrapper = component();
+    await waitOneTickAndUpdate(wrapper);
+    expect(wrapper).toBeTruthy();
+
+    await simulateComponentChange(wrapper, findRTDropdown(wrapper), InstitutionMembershipRequirement.ADDRESSES);
+
+    expect(findRTAddressError(wrapper).exists()).toBeFalsy();
+
+    // In case of a single entry which is not in the correct format
+    findRTAddressInput(wrapper).first().simulate('change', {target: {value: 'rtInvalidEmail@domain'}});
+    findRTAddressInput(wrapper).first().simulate('blur');
+    expect(findRTAddressError(wrapper).first().props().children)
+        .toBe('Following Email Addresses are not valid : rtInvalidEmail@domain');
+
+    // Multiple Email Address entries with a mix of correct (someEmail@broadinstitute.org') and incorrect format
+    findRTAddressInput(wrapper).first()
+        .simulate('change', {
+          target: {
+            value:
+                'invalidEmail@domain@org,\n' +
+                'correctEmail@someDomain.org,\n' +
+                ' correctEmail.123.hello@someDomain567.org.com   \n' +
+                ' invalidEmail   ,\n' +
+                ' justDomain.org,\n' +
+                'someEmail@broadinstitute.org\n' +
+                'nope@just#plain#wrong'
+          }
+        });
+    findRTAddressInput(wrapper).first().simulate('blur');
+    expect(findRTAddressError(wrapper).first().props().children)
+        .toBe('Following Email Addresses are not valid : invalidEmail@domain@org , invalidEmail , justDomain.org , nope@just#plain#wrong');
+
+    // Single correct format Email Address entries
+    findRTAddressInput(wrapper).first().simulate('change', {target: {value: 'correctEmail@domain.com'}});
+    findRTAddressInput(wrapper).first().simulate('blur');
+    expect(findRTAddressError(wrapper).exists()).toBeFalsy();
+  });
+
+  it('Should ignore empty string in email Domain in Controlled Tier requirement', async() => {
+    const wrapper = component();
+    await waitOneTickAndUpdate(wrapper);
+    expect(wrapper).toBeTruthy();
+
+    await simulateComponentChange(wrapper, findCTEnabled(wrapper), true);
+    await waitOneTickAndUpdate(wrapper);
+
+    await simulateComponentChange(wrapper, findCTDropdown(wrapper), InstitutionMembershipRequirement.DOMAINS);
+
+    // Single Entry with incorrect Email Domain format
```

Comment:
nit: is this comment correct? There's just an empty line, right?

Reply:
right - about half of these "incorrect format" are extraneous empty line

Reply:
updated some comments